indico/modules/events/sessions/controllers/management/__init__.py

```diff
@@ -0,0 +1,43 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request
+
+from MaKaC.webinterface.rh.conferenceModif import RHConferenceModifBase
+
+
+class RHManageSessionsBase(RHConferenceModifBase):
+    """Base RH"""
+
+    def _checkParams(self, params):
+        RHConferenceModifBase._checkParams(self, params)
+        self.event = self._conf.as_event
+
+
+class RHManageSessionBase(RHManageSessionsBase):
+    """Base RH for management of a single session"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.session
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageSessionsBase._checkParams(self, params)
+        self.session = self.event.sessions.filter_by(id=request.view_args['session_id']).first_or_404()
```

Comment:
Thanks to normalization this would work, too (and if the session id does not match the event id, the user is automatically redirected to the proper event id)

    self.session = Session.get_one(request.view_args['session_id'])